src/core/Core_error.ml

```diff
@@ -216,7 +216,10 @@ let exn_to_error rule_id file (e : Exception.t) : t =
           Exception.reraise e
       | exn ->
           let trace = Exception.to_string e in
-          let loc = Tok.first_loc_of_file file in
+          let loc =
+            if not String.(equal file "") then Tok.first_loc_of_file file
```

Comment:
maybe add a (* TODO: we should not build such Tok.t in the first place without any filename, but let's be defensive here *)

Reply:
I would write `if not (file = "") then ...`.